src/app/shared/services/loading.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoadingService {
+
+  loadingSubject = new BehaviorSubject<boolean>(false);
```

Comment:
Maybe I have missed - but have you created the handling of the loading with the help of spinner?

Reply:
yes, In the services I change the status of loading subject in the beginning and in the end of the fetching.
The subscription for loadingSubject in the core component which is rendering LoadingComponent if  loadingSubject is true